DataFormats/L1Scouting/interface/OrbitCollection.h

```diff
@@ -0,0 +1,186 @@
+  const_iterator end() const { return data_.end(); }
+
+  // iterate over elements of a bx
+  const_iterator begin(unsigned bx) const { return bxRange(bx).first; }
+  const_iterator end(unsigned bx) const { return bxRange(bx).second; }
+
+  std::pair<const_iterator, const_iterator> bxRange(unsigned bx) const {
+    if (bx > 3564)
+      throw cms::Exception("OrbitCollection::getBxVectorView")
+          << "Trying to access and object outside the orbit range. "
+          << " BX = " << bx << std::endl;
+
+    if (getBxSize(bx) > 0) {
+      return std::make_pair(data_.begin() + bxOffsets_[bx], data_.begin() + bxOffsets_[bx + 1]);
+    } else {
+      return std::make_pair(end(), end());
+    }
+  }
+
+  // get number of objects stored in a BX
+  int getBxSize(unsigned bx) const {
+    if (bx > 3564) {
+      edm::LogWarning("OrbitCollection") << "Called getBxSize() of a bx out of the orbit range."
+                                         << " BX = " << bx << std::endl;
+      return 0;
+    }
+    if (data_.empty()) {
+      edm::LogWarning("OrbitCollection") << "Called getBxSize() but collection is empty." << std::endl;
+    }
+    return bxOffsets_[bx + 1] - bxOffsets_[bx];
+  }
+
+  // get i-th object from BX
+  const T& getBxObject(unsigned bx, unsigned i) const {
+    if (bx > 3564)
+      throw cms::Exception("OrbitCollection::getBxObject") << "Trying to access and object outside the orbit range. "
+                                                           << " BX = " << bx << std::endl;
+    if (i >= getBxSize(bx))
+      throw cms::Exception("OrbitCollection::getBxObject")
+          << "Trying to get element " << i << " but for"
+          << " BX = " << bx << " there are " << getBxSize(bx) << " elements." << std::endl;
+
+    return data_[bxOffsets_[bx] + i];
+  }
+
+  // get the list of non empty BXs
+  std::vector<unsigned> getFilledBxs() const {
+    std::vector<unsigned> filledBxVec;
+    if (!data_.empty()) {
+      for (unsigned bx = 0; bx < 3565; bx++) {
+        if (getBxSize(bx) > 0)
+          filledBxVec.push_back(bx);
+      }
+    }
+    return filledBxVec;
+  }
+
+  int size() const { return data_.size(); }
+
+  T& operator[](std::size_t i) { return data_[i]; }
+  const T& operator[](std::size_t i) const { return data_[i]; }
+
+  void fillView(edm::ProductID const& id,
+                std::vector<void const*>& pointers,
+                edm::FillViewHelperVector& helpers) const {
+    edm::detail::reallyFillView(*this, id, pointers, helpers);
+  }
+
+  void setPtr(std::type_info const& toType, unsigned long index, void const*& ptr) const {
+    edm::detail::reallySetPtr<OrbitCollection<T>>(*this, toType, index, ptr);
+  }
+  void fillPtrVector(std::type_info const& toType,
+                     std::vector<unsigned long> const& indices,
+                     std::vector<void const*>& ptrs) const {
+    edm::detail::reallyfillPtrVector(*this, toType, indices, ptrs);
+  }
```

Comment:
I was wondering the point of these member functions, but I see this is an established pattern.

Could you elaborate the need for `edm::View` or `edm::Ptr` support?

Reply:
Indeed they are not needed for this work, I've removed them.